xbmc/interfaces/json-rpc/AddonsOperations.cpp

```diff
@@ -79,8 +79,18 @@ JSONRPC_STATUS CAddonsOperations::GetAddons(const std::string &method, ITranspor
     {
       if (!enabled.isBoolean())
       {
-        CAddonMgr::Get().GetAllAddons(typeAddons, false);
-        CAddonMgr::Get().GetAllAddons(typeAddons, true);
+        for (int i = ADDON_UNKNOWN+1; i < ADDON_MAX; ++i)
+        {
+          if (ADDON_REPOSITORY != (TYPE)i)
+          {
+            VECADDONS diabled;
+            if (CAddonMgr::Get().GetAddons((TYPE)i, diabled, false))
+              addons.insert(addons.end(), diabled.begin(), diabled.end());
+            VECADDONS enabled;
+            if (CAddonMgr::Get().GetAddons((TYPE)i, enabled, true))
+              addons.insert(addons.end(), enabled.begin(), enabled.end());
+          }
+        }
```

Comment:
Right. We don't really want that in JSON-RPC either because addons are identified by their addon ID and not by a combination of addon ID and extension point.